### PR TITLE
Animate transition between VideoInfo.Error and other VideoInfo states

### PR DESCRIPTION
**Pull request by strooooke**
...and some related cleanup around AddScreen

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,42 +1,16 @@
...
+# Default value: -Xmx1024m -XX:MaxPermSize=256m
 # org.gradle.jvmargs=-Xmx2048m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
-
+#
 # When configured, Gradle will run in incubating parallel mode.
 # This option should only be used with decoupled projects. More details, visit
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
 # org.gradle.parallel=true
+#Tue Sep 20 11:49:04 CEST 2022
+org.gradle.jvmargs=-Xmx2048M -Dkotlin.daemon.jvm.options\="-Xmx2048M"
```
Comment (maxhille) [inline gradle.properties:14]: please no machine specific gradle settings in the project. this belongs into `~/.gradle`

**Review by maxhille — CHANGES_REQUESTED** [review]

File: gradle.properties (hunk 1 above)
Comment (strooooke) [reply gradle.properties]: Is resolved, PTAL at the rest.

File: app/src/main/java/com/lambdasoup/watchlater/ui/add/AddScreen.kt [hunk 2]
```diff
@@ -64,11 +65,14 @@ fun AddScreen(
...
-    viewModel: AddViewModel = viewModel(),
+    onChangePlaylistClicked: () -> Unit,
+    onAbortChangePlaylist: () -> Unit,
+    onSelectPlaylist: (YoutubeRepository.Playlists.Playlist) -> Unit,
+    viewModel: LiveData<AddViewModel.Model>,
 ) {
     WatchLaterTheme {
         @Suppress("UNCHECKED_CAST")
-        val viewState = viewModel.model.observeAsState() as State<AddViewModel.Model>
+        val viewState = viewModel.observeAsState() as State<AddViewModel.Model>
```
Comment (maxhille) [inline app/src/main/java/com/lambdasoup/watchlater/ui/add/AddScreen.kt:79]: Is this not possible without an unchecked cast?

**Review by maxhille — APPROVED** [review]

File: app/src/main/java/com/lambdasoup/watchlater/ui/add/AddScreen.kt (hunk 2 above)
Comment (strooooke) [reply app/src/main/java/com/lambdasoup/watchlater/ui/add/AddScreen.kt:79]: I don't think so. `viewModel` is a `LiveData`, the extension function there that converts to `MutableState` knows that LiveData values are nullable in Java, so it insists on the resulting state having nullable value.

_We_ know that this LiveData won't have non-null value.

Alternative would be to `!!` on each usage. I think that the unchecked cast is cleaner and clearer.

**Pull request merged**